### PR TITLE
Add blog post for restic 0.17.0 release

### PR DESCRIPTION
**Pull request by rawtaz**
This PR needs to be rebased after the rest-server PR #38 has been merged.

### CODE REVIEW

File: content/blog/restic-0.17.0-released.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- Resumable and optimized [`prune`](https://restic.readthedocs.io/en/stable/060_forget.html) command, now using up to 60% less memory and being less write-intensive.
+- Ability to [skip snapshot creation if nothing changed](https://restic.readthedocs.io/en/stable/040_backup.html#skip-creating-snapshots-if-unchanged).
+- Support for [repositories with empty password](https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html#repositories-with-empty-password).
+- Support for [FUSE-T on macOS](https://restic.readthedocs.io/en/stable/050_restore.html#restore-using-mount).
+- Support for [size](https://restic.readthedocs.io/en/latest/045_working_with_repos.html#listing-all-snapshots) and [summary](https://restic.readthedocs.io/en/latest/075_scripting.html#summary) statistics in snapshots.
```
Comment (MichaelEischer) [inline content/blog/restic-0.17.0-released.md]: These links should also use `stable` instead of `latest`.

**Review by MichaelEischer — COMMENTED** [review]
Just one minor nitpick. I wonder, whether we should use links that contain a specific version to make sure they won't break in the future. That is for example https://restic.readthedocs.io/en/v0.17.0/060_forget.html instead of https://restic.readthedocs.io/en/stable/060_forget.html

File: content/blog/restic-0.17.0-released.md (hunk 1 above)
Comment (rawtaz) [reply content/blog/restic-0.17.0-released.md]: Fixed, thanks for noticing!

**Comment by rawtaz** [comment]
> Just one minor nitpick. I wonder, whether we should use links that contain a specific version to make sure they won't break in the future. That is for example https://restic.readthedocs.io/en/v0.17.0/060_forget.html instead of https://restic.readthedocs.io/en/stable/060_forget.html

After discussing this we decided to keep the links pointing to stable instead of the specific version. The reasoning is that the risk of the links/anchors changing/breaking is quite small, and would be detected by the link checker when tests for the site runs (which happens at least whenever we update the site, and could also be automated if need be). There's greater value in the reader always being linked to the latest updates of the docs for a given feature, than what the risk of the links breaking is.

**Comment by rawtaz** [comment]
Just noticed that the links to the GitHub release page were missing the canonical `tag/` path component in their URLs, so I updated that for this 0.17.0 blog post.

**Pull request merged**